### PR TITLE
add tests for GDT audit log

### PR DESCRIPTION
**Pull request by zkrejcov**
... and clean up AuditLogEntryCellHelper a tiny bit.

This PR relies (a bit) on https://github.com/droolsjbpm/drools/pull/820.

(part n+1 of moving-tests-to-community series)

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+
+    @Mock
+    private DateTimeFormat format;
+    private String labelClass = "mockCssClassForLabel";
+    private AuditLogEntryCellHelper helper;
+
+    @Before
+    public void setUp() throws NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+        when(format.format(any(Date.class))).thenReturn("mockDateString");
+        // This also mock AuditLogEntryCellHelper.Template, which then for each method call returns "<mathod name>(<call arguments>)
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java]: Typo (if I was being picky) ```<m*ethod name>```

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+        Long l = new Long(123);
+        assertEquals(l.toString(), helper.convertValueToString(l));
+        Short s = 123;
+        assertEquals(s.toString(), helper.convertValueToString(s));
+        assertEquals("surprise!", helper.convertValueToString("surprise!"));
+        assertEquals(this.toString(), helper.convertValueToString(this));
+    }
+
+    @Override
+    public String toString() {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java]: Is this needed?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import org.kie.workbench.common.services.datamodel.model.PackageDataModelOracleBaselinePayload;
+import org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracle;
+import org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleFactory;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.uberfire.backend.vfs.Path;
+
+@WithClassesToStub({Text.class, DateTimeFormat.class})
+@RunWith(GwtMockitoTestRunner.class)
+public class GuidedDecisionTablePresenter_AuditLogTest {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java]: It's not merged (yet) but https://github.com/droolsjbpm/drools-wb/pull/218 moves setup of ```GuidedDecisionTablePresenter``` to a base super class for tests.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        verify(synchronizer, times(4)).updateColumn(any(BaseColumn.class), any(BaseColumn.class));
+        verify(synchronizer).updateColumn(any(Pattern52.class), any(ConditionCol52.class), any(Pattern52.class), any(ConditionCol52.class));
+        verify(model, times(5)).getAuditLog();
+        assertEquals(5, model.getAuditLog().size());
+        for (UpdateColumnAuditLogEntry entry : model.getAuditLog().toArray(new UpdateColumnAuditLogEntry[0])) {
+            assertEquals(diffs, entry.getDiffs());
+        }
+    }
+
+    public class MockEvent implements Event {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java]: There's a ```EventSourceMock``` in Uberfire that does this.. could you please use that instead?

**Comment by manstis** [comment]
@zkrejcov Thanks... just a couple of comments.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java (hunk 2 above)
Comment (zkrejcov) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java]: It seemed to be a nice/easy way to test that `helper.convertValueToString(..)` doesn't blow up on unexpected types (see the original code for the 'expected' types). I can remove the check and the `toString()` though, no problem.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java (hunk 3 above)
Comment (zkrejcov) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java]: Hmm, I see. But the `initialiseAuditLog()` of the presenter is overridden there too, which is not very helpful for this test. That's actually why I created a new test class and didn't rather add to the already existing `GuidedDecisionTablePresenterTest`. Any ideas regarding this tidbit..?

Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter_AuditLogTest.java]: Fair enough; if your test sets things up differently then so be it... no point in reusing (the new base test class) if you cannot!

**Comment by zkrejcov** [comment]
Thanks for checking @manstis - the issues you pointed out should now be gone.

jenkins please retest this

**Comment by manstis** [comment]
Thanks @zkrejcov 

+1 to merge.

**Comment by Rikkola** [comment]
@manstis @zkrejcov I think I have a different JDK or different version of Java. So testUpdate1, testUpdate2 and testUpdate3 fail locally. Going to check what the issue is since I can reproduce it.

**Comment by Rikkola** [comment]
@manstis @zkrejcov Ok, my mistake. It is also broken in master.
https://kie-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/master/job/drools-wb/1039/

**Comment by zkrejcov** [comment]
Ah...as I stated in my first comment/description, it needs the drools PR / https://github.com/droolsjbpm/drools/pull/820.

**Comment by Rikkola** [comment]
Ok thanks, strange it should work then. Both are in.

**Comment by zkrejcov** [comment]
The drools one is not as far as I can see.

**Comment by csadilek** [comment]
@zkrejcov @Rikkola my fault, missed the other one. Merging now...

**Comment by Rikkola** [comment]
No problem. 
@zkrejcov You were right. I took a quick look and I think the "merged" I saw in the drools PR was just the reference to this PR.

**Pull request merged**